CMakeLists.txt

```diff
@@ -108,6 +108,25 @@ if(ENABLE_JIMPLE_FRONTEND)
     add_definitions(-DENABLE_JIMPLE_FRONTEND)
 endif()
 
+if(ENABLE_PYTHON_FRONTEND)
+    # Command to check if ast2json is installed
+    execute_process(
+        COMMAND python3 -c "import ast2json"
+        RESULT_VARIABLE PYTHON_PACKAGE_RESULT
+        OUTPUT_QUIET
+    )
+
+    # Install ast2json if not available
+    if (NOT PYTHON_PACKAGE_RESULT EQUAL 0)
+        message("Installing ast2json")
+        execute_process(
+            COMMAND pip3 install ast2json
```

Comment:
Can we put this behind the `DOWNLOAD_DEPENDENCIES` CMake flag?

Reply:
"DOWNLOAD_DEPENDENCIES" is set to "On" only when using "script/build.sh". This can be an issue when building directly with CMake (as outlined in BUILDING.md).

Reply:
This suggestion is independent of build.sh. `DOWNLOAD_DEPENDENCIES` is a user-facing option. I usually have it disabled, for reasons. In case the PYTHON_PACKAGE_RESULT is not 0 and DOWNLOAD_DEPENDENCIES is turned off, there should be no downloading of dependencies.

As a side note: `pip install` installs either
- system-wide (e.g. under `/`) in case it has rights to do so, or
- local to the current user under `$HOME/.local`.

Both are not the case with DOWNLOAD_DEPENDENCIES as we have it now: it only does it in the local cmake build directory. @rafaelsamenezes Do you have a suggestion how we could work around this issue?

Reply:
> This suggestion is independent of build.sh
Yes, we can use -DDOWNLOAD_DEPENDENCIES independently of build.sh, however this option is not mentioned in the build documentation (BUILDING.md), which could pose difficulties for newcomers.
Should we update it?

Reply:
Yes, but doesn't have to be this PR.

Reply:
> Both are not the case with DOWNLOAD_DEPENDENCIES as we have it now: it only does it in the local cmake build directory. @rafaelsamenezes Do you have a suggestion how we could work around this issue?

I would prefer to keep all side effects of `DOWNLOAD_DEPENDENCIES` to inside the build directory. If we need the dependency for building purposes (e.g. "python2goto")...  we could [set up a python virtual environment](https://discourse.cmake.org/t/possible-to-create-a-python-virtual-env-from-cmake-and-then-find-it-with-findpython3/1132) inside the build folder.

As a runtime dependency: I don't think it is the CMake responsability to install it as this will not break sharing builds between systems. IMO when trying to parse Python, do a check if the package is available. If not then abort with an error message containing the pip install command. This shuld keep common users happy and advanced users will know how to adapt the command.

Reply:
Agreed, I will remove this part from the CMakeLists and handle it in Python.

Reply:
Done